lib/algos/list/delimiterMaster.js

```diff
@@ -122,11 +120,13 @@ class DelimiterMaster extends Delimiter {
              *   when an entry is a PHD avoids the skip on version for the
              *   next entry). In that case we expect the master version to
              *   follow. */
+
             if (key === this.prvPHDKey) {
                 this.prvKey = undefined;
                 return FILTER_ACCEPT;
             }
             this.prvKey = key;
+            this[this.nextContinueMarker] = key;
```

Comment:
the key changes are setting the nextContinuationToken to key instead of commonPrefix here and on line 179.